src/Storefront/Theme/Validator/SCSSValidator.php

```diff
@@ -212,4 +203,86 @@ private static function extractSCSSvars(string $value): array
 
         return $vars;
     }
+
+    private static function isHex(string $hexCode): bool
+    {
+        preg_match('/^[a-f0-9]*$/i', $hexCode, $parsed);
+
+        return isset($parsed[0]) && $parsed[0] === $hexCode;
+    }
+
+    /**
+     * Validates if the given HSL code is valid.
+     *
+     * It supports both classic and modern syntax.
+     * Will forward to isHSLA for classic syntax with 'hsla'.
+     */
+    private static function isHSL(string $hslCode): bool
+    {
+        if (!str_starts_with($hslCode, 'hsl')) {
+            return false;
+        }
+
+        if (str_starts_with($hslCode, 'hsla')) {
+            return self::isHSLA($hslCode);
+        }
+
+        $hue = '(?:360(?:\.0+)?|3[0-5]\d(?:\.\d+)?|[12]\d{2}(?:\.\d+)?|[1-9]?\d(?:\.\d+)?)';
+        $percent = '(?:100|[1-9]?\d)%';
+        $alpha = '(?:0?\.\d+|1(?:\.0+)?|0|(?:100|[1-9]?\d)%)';
+
+        // Modern: hsl(h s% l% / a?)  (spaces; optional / alpha)
+        $patternModern = '/^hsl\(\s*' . $hue . '(?:deg)?\s+' . $percent . '\s+' . $percent . '(?:\s*\/\s*' . $alpha . ')?\s*\)$/i';
+
+        // Legacy: hsl(h, s%, l%)  (commas; no alpha)
+        $patternLegacy = '/^hsl\(\s*' . $hue . '(?:deg)?\s*,\s*' . $percent . '\s*,\s*' . $percent . '\s*\)$/i';
+
+        return preg_match($patternModern, $hslCode) === 1 || preg_match($patternLegacy, $hslCode) === 1;
+    }
+
+    /**
+     * Validates if the given HSLA code is valid.
+     */
+    private static function isHSLA(string $hslaCode): bool
+    {
+        $hue = '(?:360(?:\.0+)?|3[0-5]\d(?:\.\d+)?|[12]\d{2}(?:\.\d+)?|[1-9]?\d(?:\.\d+)?)';
+        $percent = '(?:100|[1-9]?\d)%';
+        $alpha = '(?:0?\.\d+|1(?:\.0+)?|0|(?:100|[1-9]?\d)%)';
+
+        // Legacy HSLA: hsla(h, s%, l%, a)
+        $patternHsla = '/^hsla\(\s*' . $hue . '(?:deg)?\s*,\s*' . $percent . '\s*,\s*' . $percent . '\s*,\s*' . $alpha . '\s*\)$/i';
+
+        return preg_match($patternHsla, $hslaCode) === 1;
+    }
+
+    /**
+     * Validates if the given RGB code is valid.
+     *
+     * It supports both classic and modern syntax.
+     * Will forward to isRGBA for classic syntax with 'rgba'.
+     */
+    private static function isRGB(string $rgbCode): bool
+    {
+        if (!str_starts_with($rgbCode, 'rgb')) {
+            return false;
+        }
+
+        if (str_starts_with($rgbCode, 'rgba')) {
+            return self::isRGBA($rgbCode);
+        }
+
+        $patternRgb = '/^rgb\(\s*(?:(?:25[0-5]|2[0-4]\d|1?\d?\d|(?:100|[1-9]?\d)%)\s+(?:25[0-5]|2[0-4]\d|1?\d?\d|(?:100|[1-9]?\d)%)\s+(?:25[0-5]|2[0-4]\d|1?\d?\d|(?:100|[1-9]?\d)%)(?:\s*\/\s*(?:0?\.\d+|1(?:\.0+)?|0|(?:100|[1-9]?\d)%))?|(?:25[0-5]|2[0-4]\d|1?\d?\d|(?:100|[1-9]?\d)%)\s*,\s*(?:25[0-5]|2[0-4]\d|1?\d?\d|(?:100|[1-9]?\d)%)\s*,\s*(?:25[0-5]|2[0-4]\d|1?\d?\d|(?:100|[1-9]?\d)%))\s*\)$/i';
```

Comment:
<img src="https://media.tenor.com/DGUfkBTw4f4AAAAM/shocked-sunglasses.gif">

Reply:
It supports modern and classic syntax. 😅

Reply:
We have links like this in code to explain it:
https://regex101.com/r/idIfbk/1